gradoop-data-integration/src/test/java/org/gradoop/dataintegration/transformation/VertexDeduplicationTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2014 - 2019 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.dataintegration.transformation;
+
+import org.gradoop.common.model.impl.pojo.EPGMVertex;
+import org.gradoop.flink.model.GradoopFlinkTestBase;
+import org.gradoop.flink.model.impl.epgm.LogicalGraph;
+import org.gradoop.flink.util.FlinkAsciiGraphLoader;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test for the {@link VertexDeduplication} operator.
+ */
+public class VertexDeduplicationTest extends GradoopFlinkTestBase {
+
+  /**
+   * Run the operator on a test graph.
+   *
+   * @throws Exception when the execution in Flink fails.
+   */
+  @Test
+  public void testWithGraph() throws Exception {
+    FlinkAsciiGraphLoader loader = getLoaderFromString("input[" +
+      "(:a {key: 1L})-[:e {att: 1L}]->(:a {key: 1L})-[:e {att: 2L}]->(:b {key: 1L})" +
+      "-[:e {att: 3L}]->(:a {key: 2L})(:a {key: 2L})" +
+      "]" +
```

Comment:
This example is too simple. You can test it in one graph, but at least add multiple keys and different not duplicated vertices (e.g. with one property equal and one missing or the same key but different values/types, etc).

Reply:
I changed it to a more complex example. Cases for all missing keys etc. are tested with the key extractor function